### ISSUE TITLE
Geography / Landscape

### ISSUE THREAD

**Issue by kylerschin** [decentraland/proposals#11]
Hello! I suggest that there be a natural landscape, including plains, mountains, and occasional large bodies of water. We can then assign landscapes based off an algorithm based off a "biome". We then generate random sets of rocks, grass, trees, terrain, water as see fit. This allows for more creativity with objects such as tropical beaches or hikes in the jungle, climbing mountains, house in the forest, etc. People then can modify the landscape as they see fit. Hopefully, this is better than our current infinite "desert".

**Comment by dwrowley** [comment]
Macroverse.io is another blockchain-based virtual world initiative that uses procedurally generated terrain - they don't see to be as far along as the DCL team though, but perhaps a procedurally generated terrain model could be added into DCL.

**Comment by kylerschin** [comment]
Hmmm... Marcoverse makes the user pay too. Sad!
I would like that. My little sister (she is 11 yrs old) said that she will buy a bunch of mana and make a huge jungle island with waterfalls and nature stuff. Then invite all her friends over.
Thats how I came up with it. Users can explore for the perfect plot, then claim it. Terrain is awesome.
@dwrowley

**Comment by iSasFTW** [comment]
I love the idea of terrain, but there's the problem of  execution. Since terrain would have to be the default model of your tile, you would be able to change it to whatever you want. This means that if you have height differences across tiles, the adjacent tile's owner could easily switch the 3D model, effectively ruining the aesthetic. Terrain could work if you have a list of models generated/created in advance, that are tilable. Basically the sides of the tiles will always be at the lowest possible height, but the

**Comment by kylerschin** [comment]
@iSasFTW I think map builders could simply take advantage of large amounts of flat land, i.e. Plains, deserts, beaches. However, some variation is necessary for a beautiful and meaningful experience.

**Comment by iSasFTW** [comment]
@kylerschin Yeah, definitely. The real problem are the trolls, although I hope the price of land will discourage them.

**Comment by kylerschin** [comment]
@iSasFTW Describe Trolls....

**Comment by iSasFTW** [comment]
@kylerschin People who intentionally want to ruin the aesthetic or simply just annoy others. It's the internet, so there's bound to be people like that. Also people who only hold land for value/just for the sake of having it, but never upload any meaningful content. They're not trolls per se, but have a similar effect on the scenery/value of surrounding tiles.

**Comment by kylerschin** [comment]
Oh. I understand. Let's say someone buys a tile, then fills it with random blocks and inappropriate symbols, hate speech, or things intended to bully or harass others. Or dig a giant pit and make a giant tower that fills the entire grid.

**Comment by kylerschin** [comment]
I guess the only way is just to go out to some far out place and buy a bunch of plots next to each other. :(

**Comment by pookage** [comment]
Whilst the obvious goal of decentraland is to be decentralised, every fictional metaverse has fundamental base rules - ie. you can't interact with someone's possessions without their permission etc.  I'm in agreement that the terrain should be interesting in and of itself - the best art is built around constraints; nobody will build bridges unless there are rivers to cross.

SO, given that, I don't see any reason why we shouldn't normalise the edges of adjacent land to maintain a natural and contiguous feel.  Add it into the fundamental rules of the universe - the edges of your LAND will be normalised to match nearby LAND, and you cannot terraform the edges of your LAND in any case where it will affect adjacent LAND without their permission. 

We could add a collaboration mechanic where adjacent LAND owners need to work together if they want to adjust LAND that borders each other's plots?

Thoughts?

**Comment by kylerschin** [comment]
@pookage people could simply add code that prevents crap on their plot.

**Comment by pookage** [comment]
@kylerschin - I was proposing a way to ensure the terrain was contiguous, unless you were saying why it wouldn't work?

**Comment by dwrowley** [comment]
What do you all think about the current size of a LAND plot? (10m x 10m / 33ft x 33ft) - seems on the small side to me... (which makes have visual continuity between plots even harder)

**Comment by pookage** [comment]
Agreed that it seems small - but [this article from the blog](https://blog.decentraland.org/building-a-land-economy-with-the-mana-token-ccb8d0d5e010) indicates that land would be 100m x 100m, not 10m x 10m, so plenty to play with there - and also plenty of room to tweak your terrain if we do have contiguous landscapes.

**Comment by kylerschin** [comment]
@pookage hmmmm.....

**Comment by dwrowley** [comment]
The blog says it is 100 square meters - which would be 10m x 10m ... not 100 meters squared.

**Comment by pookage** [comment]
Aah, my bad.  Good shout.

**Comment by kylerschin** [comment]
@pookage @dwrowley What do you think about oceans, islands, and other water stuff like rivers and lakes, waterfalls and stuff like that?

**Comment by pookage** [comment]
@kylerschin - I think that different types of terrain lends itself to different activities; water could open up boating as a transport and diving as an activity - just as mountains could open up skiing and climbing.  Plus the more natural aesthetic diversity - the better as far as I'm concerned!

It looks like [some people have already created some good work along these lines](https://github.com/bryik/aframe-terrain-model-component) in A-Frame as well, which gives us a good starting point...

**Comment by dwrowley** [comment]
it will be interesting to see how this works out...  we have to set the right balance between giving the LAND owner control over their plot, while making the overall experience as compelling for visitors as possible (having a good browsing experience while moving from LAND to LAND, good continuity, interesting and unique domains).   Procedurally generated terrain is a good idea, I believe, but that will also make certain coordinates more attractive than others (which is fine).  

If the team wants to do procedurally generated terrain, they need to get to it quickly, before the genesis block, since doing it afterwards would be problematic.  I'm not sure I see that happening by then.

**Comment by pookage** [comment]
haha, well, in the event of procedural land not occurring at genesis - consider this my vote for joining a collective of LAND owners willing to run some generation on our own combined plots 😆

**Comment by dwrowley** [comment]
There are a number of impressive terrain generators out there, like this one: http://www.world-creator.com/home-tmp/

(has an integration w. Unity too)
